### ISSUE TITLE
No metrics for `HttpClient` are created when tracing is enabled

### ISSUE THREAD

**Issue by pyohannes** [open-telemetry/opentelemetry-dotnet-instrumentation#2834]
# Bug Report

## Symptom

**Describe the bug**
When using version v1.0.0-rc.2 and both metrics and traces are enabled, no metrics from the `OpenTelemetry.Instrumentation.Http` instrumentation library are created. Traces are created.

When traces are disabled (`OTEL_DOTNET_AUTO_TRACES_ENABLED=false`) HTTP metrics are created. 

However, it seems there's no way to export both metrics and traces.

**Runtime environment (please complete the following information):**

- OpenTelemetry Automatic Instrumentation version: 1.0.0-rc.2
- OS: Ubuntu 22
- .NET version: .NET 7

**Additional context**
After some debugging, I think that's what's happening:
* HTTP metrics instrumentation is [supposed to be initialized when `System.Net.Http` is loaded](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/src/OpenTelemetry.AutoInstrumentation/Loading/Initializers/HttpClientMetricsInitializer.cs#L25).
* The `HttpClientMetricsInitializer` mentioned above is registered during a call to [MeterProviderBuilder.UseEnvironmentVariables](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs#L142).
* However, `System.Net.Http` is already loaded before when the [traces pipeline is built](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs#L123), likely by initializing exporters for tracing or HTTP instrumentation for tracing.

TLDR; At the moment when `HttpClientMetricsInitializer` registers its event handler that should fire when the `System.Net.Http` assembly is loaded, the assembly is already loaded.

## Reproduce

 1. Set environment variables to use .NET auto-instrumentation, enable both traces and metrics with the default OTLP exporter.
 2. Also export to the console:
   ```
   OTEL_DOTNET_AUTO_TRACES_CONSOLE_EXPORTER_ENABLED=true  
   OTEL_DOTNET_AUTO_METRICS_CONSOLE_EXPORTER_ENABLED=true
   ```
 3. Run an example application that makes an HTTP request using `HttpClient`.
 4. No HTTP metrics are exported.

**Comment by pyohannes** [comment]
[This branch](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/compare/main...pyohannes:opentelemetry-dotnet-instrumentation:split-builder-init?expand=1) contains a prototype for a fix, which works for me. Building the `TracerProviderBuilder` is delayed until the `MeterProviderBuilder` is configured.

If you'd like and think this fix is feasible, I can work on a PR.

**Comment by RassK** [comment]
@pyohannes thanks! seems quite a severe bug. I opened a fix PR.

**Comment by rajkumar-rangaraj** [comment]
@RassK Do we understand why this issue only began after RC2, and not before?

**Comment by pyohannes** [comment]
> @RassK Do we understand why this issue only began after RC2, and not before?

I'm pretty sure I could reproduce this issue with pre-RC2 versions. I can try again, if it's of any use to you.

**Comment by Kielek** [comment]
@pyohannes, it will be great if you can reproduce on some older version.
